File: Java/DeepspaceJava/deepspace/test/deepspace/TestP1.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author paula
 */
public class TestP1 {
    public static void main(String[] args) {
        Loot botin;
        SuppliesPackage supplies;
        ShieldBooster sBooster;
        Weapon arma1, arma2, arma3;
        
        // Loot:
        int i = 0;

        while (i < 3){
            botin = new Loot(i, i, i, i, i);

            System.out.format("\nBotin %d",i+1);
            System.out.format("\n  Numero de suministros: %d", botin.getNSupplies());
            System.out.format("\n  Numero de armas: %d", botin.getNWeapons());
            System.out.format("\n  Numero de escudos: %d", botin.getNShields());
            System.out.format("\n  Numero de hangares: %d", botin.getNHangars());
            System.out.format("\n  Numero de medallas: %d", botin.getNMedals());

            i += 1;
        }

        System.out.format("\n");

        // Supplies
        i = 0;

        while (i < 3){
            supplies = new SuppliesPackage(i, i, i);

            System.out.format("\nSuministro %d", i+1);
            System.out.format("\n Armamento: %f", supplies.getAmmoPower());
            System.out.format("\n Combustible: %f", supplies.getFuelUnits());
            System.out.format("\n Energia para los escudos: %f", supplies.getShieldPower());

            i += 1;
        }

        // ShieldBooster
        i = 0;

        while (i < 3){
            sBooster = new ShieldBooster("Escudo", i, i);

            System.out.format("\nShield Booster %d", i+1);
            System.out.format("\n %s", sBooster.getName());
            System.out.format("\n Boost: %f", sBooster.getBoost());
            System.out.format("\n Usos: %d", sBooster.getUses());

            i += 1;
        }

        // Weapon

        arma1 = new Weapon("laser", WeaponType.LASER, 1);
        arma2 = new Weapon("misil", WeaponType.MISSILE, 2);
        arma3 = new Weapon("plasma", WeaponType.PLASMA, 0);

        System.out.format("\n \nArma 1:");
        System.out.format("\nTipo: %s", arma1.getType());
        System.out.format("\n Usos: %d", arma1.getUses());
        System.out.format("\n Potencia: %f", arma1.power());
        System.out.format("\n Metodo useIt: ");
        System.out.format("\n %f", arma1.useIt());
        System.out.format("\n %f", arma1.useIt());
        System.out.format("\n \nArma 2:");
        System.out.format("\nTipo: %s", arma2.getType());
        System.out.format("\n Usos: %d", arma2.getUses());
        System.out.format("\n Potencia: %f", arma2.power());
        System.out.format("\n Metodo useIt: ");
        System.out.format("\n %f", arma2.useIt());
        System.out.format("\n %f", arma2.useIt());
        System.out.format("\n \nArma 3:");
        System.out.format("\nTipo: %s", arma3.getType());
        System.out.format("\n Usos: %d", arma3.getUses());
        System.out.format("\n Potencia: %f", arma3.power());
        System.out.format("\n Metodo useIt: ");
        System.out.format("\n %f", arma3.useIt());
        System.out.format("\n %f", arma3.useIt());
    }
}

File: Ruby/DeepSpaceRuby/test/test_p1.rb
#encoding:utf-8

require '../lib/CombatResult.rb'
require '../lib/Dice.rb'
require '../lib/GameCharacter.rb'
require '../lib/Loot.rb'
require '../lib/ShieldBooster.rb'
require '../lib/ShotResult.rb'
require '../lib/SuppliesPackage.rb'
require '../lib/Weapon.rb'
require '../lib/WeaponType.rb'

module Pruebas

  class TestP1

    def initialize

    end

    def self.main

      # Loot:
      i = 0

      while i < 3
        botin = Deepspace::Loot.new(i, i, i, i, i)

        print "\nBotin ",i+1
        print "\n  Numero de suministros: ", botin.nSupplies
        print "\n  Numero de armas: ", botin.nWeapons
        print "\n  Numero de escudos: ", botin.nShields
        print "\n  Numero de hangares: ", botin.nHangars
        print "\n  Numero de medallas: ", botin.nMedals

        i += 1
      end

      print "\n"

      # Supplies
      i = 0

      while i < 3
        supplies = Deepspace::SuppliesPackage.new(i, i, i)

        print "\nSuministro ", i+1
        print "\n Armamento: ", supplies.ammoPower
        print "\n Combustible: ", supplies.fuelUnits
        print "\n Energia para los escudos: ", supplies.shieldPower

        i += 1
      end

      # ShieldBooster
      i = 0

      while i < 3
        sBooster = Deepspace::ShieldBooster.new("Escudo", i, i)

        print "\nShield Booster ", i+1
        print "\n ", sBooster.name
        print "\n Boost: ", sBooster.boost
        print "\n Usos: ", sBooster.uses

        i += 1
      end

      # Weapon

      arma1 = Deepspace::Weapon.new("laser", Deepspace::WeaponType::LASER, 1)
      arma2 = Deepspace::Weapon.new("misil", Deepspace::WeaponType::MISSILE, 2)
      arma3 = Deepspace::Weapon.new("plasma", Deepspace::WeaponType::PLASMA, 0)

      puts "\n \nArma 1:"

      if arma1.type == Deepspace::WeaponType::LASER then
        print "\nTipo: laser"
      elsif arma1.type == Deepspace::WeaponType::MISSILE then
        print "\nTipo: misil"
      elsif arma1.type == Deepspace::WeaponType::PLASMA then
        print "\nTipo: plasma"
      end

      print "\n Usos: ", arma1.uses
      print "\n Potencia: ", arma1.power
      print "\n Metodo useIt: "
      print "\n ", arma1.useIt
      print "\n ", arma1.useIt

      puts "\n \nArma 2:"

      if arma2.type == Deepspace::WeaponType::LASER then
        print "\nTipo: laser"
      elsif arma2.type == Deepspace::WeaponType::MISSILE then
        print "\nTipo: misil"
      elsif arma2.type == Deepspace::WeaponType::PLASMA then
        print "\nTipo: plasma"
      end

      print "\n Usos: ", arma2.uses
      print "\n Potencia: ", arma2.power
      print "\n Metodo useIt: "
      print "\n ", arma2.useIt
      print "\n ", arma2.useIt

      puts "\n \nArma 3:"

      if arma3.type == Deepspace::WeaponType::LASER then
        print "\nTipo: laser"
      elsif arma3.type == Deepspace::WeaponType::MISSILE then
        print "\nTipo: misil"
      elsif arma3.type == Deepspace::WeaponType::PLASMA then
        print "\nTipo: plasma"
      end

      print "\n Usos: ", arma3.uses
      print "\n Potencia: ", arma3.power
      print "\n Metodo useIt: "
      print "\n ", arma3.useIt
      print "\n ", arma3.useIt
    end

  end

end

Pruebas::TestP1.main
